test/validator.h

```diff
@@ -30,26 +35,48 @@ void testValidate(
     const ValidationConstants& tolerances = ValidationConstants()) {
   FusionGuard fg(fusion);
 
+  std::vector<Val*> non_hidden_outputs;
+  std::copy_if(
+      fusion->outputs().begin(),
+      fusion->outputs().end(),
+      std::back_inserter(non_hidden_outputs),
+      [fusion](Val* out) {
+        // Returns true when `out` is **not** an aliased output that's hidden
+        // from integration. Hidden outputs won't show up in `fusion_outputs`
+        // for us to compare, so we skip them.
+        const AliasInfo* alias_info = fusion->getOutputAlias(out).second;
+        if (alias_info == nullptr) {
+          return true;
+        }
+        return !alias_info->hide_output;
+      });
+
   auto expr_eval = bindInputsAndLaunchParams(fusion, aten_inputs, lparams);
 
   auto reduction_sizes =
       ReductionSizeMapper::computeReductionSizes(fusion, expr_eval);
 
   if (aten_outputs.empty()) {
-    for (auto v : fusion->outputs()) {
-      aten_outputs.emplace_back(expr_eval.evaluate(v).as<at::Tensor>());
+    for (Val* out : non_hidden_outputs) {
+      aten_outputs.emplace_back(expr_eval.evaluate(out).as<at::Tensor>());
```

Comment:
Only add non-hidden outputs so the check at line 66 passes.

Reply:
fixes issue #1596